homeassistant/components/mqtt/__init__.py

```diff
@@ -735,6 +717,15 @@ async def finish_dump(_):
     if conf.get(CONF_DISCOVERY):
         await _async_setup_discovery(hass, conf, entry)
 
+    if DATA_MQTT_RELOAD_NEEDED in hass.data:
+        hass.data.pop(DATA_MQTT_RELOAD_NEEDED)
+        await hass.services.async_call(
+            DOMAIN,
+            SERVICE_RELOAD,
+            {},
+            blocking=False,
+        )
+
```

Comment:
I discussed this with @MartinHjelmare a bit, he suggests to unconditionally load manually configured platforms when setting up the config entry instead.